Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepFindBlocks.cs

```diff
@@ -35,16 +38,23 @@ public BlockStoreInnerStepFindBlocks(ILoggerFactory loggerFactory)
         {
             this.logger.LogTrace("()");
 
-            while (await ShouldStopFindingBlocks(context) == false)
+            var batchSize = BlockStoreInnerStepContext.DownloadStackThreshold - context.DownloadStack.Count;
+            var batchList = new List<ChainedBlock>(batchSize);
+			while (batchList.Count < batchSize)
             {
+	            if (await ShouldStopFindingBlocks(context))
+	            {
+					context.StopFindingBlocks();
+	                break;
+	            }
+
+                batchList.Add(context.NextChainedBlock);
                 context.DownloadStack.Enqueue(context.NextChainedBlock);
                 context.GetNextBlock();
             }
 
-            context.StopFindingBlocks();
```

Comment:
Should'nt we always STOP to ask for blocks until a stop condition has been found or the download stack threshold reached?